src/common.js

```diff
@@ -57,18 +58,28 @@ exports.error = error;
 //@
 //@ Examples:
 //@
-//@ ```
+//@ ```javascript
 //@ var foo = ShellString('hello world');
 //@ ```
 //@
 //@ Turns a regular string into a string-like object similar to what each
 //@ command returns. This has special methods, like `.to()` and `.toEnd()`
-var ShellString = function (str, stderr) {
-  var that = new String(str);
-  that.to = wrap('to', _to, {idx: 1});
-  that.toEnd = wrap('toEnd', _toEnd, {idx: 1});
-  that.stdout = str;
+var ShellString = function (stdout, stderr) {
+  var that;
+  if (stdout instanceof Array) {
+    that = stdout;
+    that.stdout = stdout.join('\n')+'\n';
+  } else {
+    that = new String(stdout);
+    that.stdout = stdout;
+  }
   that.stderr = stderr;
+  that.to    = function() {wrap('to', _to, {idx: 1}).apply(that.stdout, arguments); return that;};
+  that.toEnd = function() {wrap('toEnd', _toEnd, {idx: 1}).apply(that.stdout, arguments); return that;};
```

Comment:
Not sure of a good way to do this, because if I move it out of the function, then `that` isn't defined anymore. So then the code only works for strings, not arrays as well. So I think we might have to leave it as-is. The code should still be very efficient, since `wrap` is just a closure.

Or we can modify `.to()` to `join` `this` if it's an array, but that would incur a performance hit.